### DISCUSSION TITLE
strange readings

### DISCUSSION THREAD

**Discussion by exxico** [klaasnicolaas/home-assistant-glow#153] [Show and tell]
i'm use this - and voila! all working! 
https://github.com/esphome/issues/issues/2823

____________________________________________________
Yo!

Got walkaround!!! It's a bit ugly, but...

So idea is that pulse_meter component tracks both rising and falling edges of the pulse (meanwhile doc says that only rising one). Thus, while not fixed, we need some pulse divider to divide sensor pulse count by two and put result as pulse_meter input. We can make divider hardware with couple of triggers or software (if you have enough free pins)

Here is my config:

binary_sensor:
  - platform: gpio
    id: water_count_raw
    pin: D0 #connect you water, electricity, etc. counter here
    on_release: # pulse edge doesn't matter, but only one of two !!!
      then:
        - switch.toggle: ipin

switch:
  - platform: gpio
    id: ipin
    pin: D1
    
sensor:
  - platform: pulse_meter
    pin: D2
    name: 'Water Speed'

Connect sensor output to DO and shorten D1 to D2.
That's it! No internal_filter needed, nothing) Just working)

**Comment by exxico** [comment]
I put a 1000 microfarad capacitor on the power supply and the glitches went away.

**Comment by markhss** [comment]
I am having issues with strange readings on my pulse sensor for my electricity meter.

The light sensing board always flashes to say it has detected a pulse, but under test conditions (with a flashing light rather than the electricity meter) it will continue to flash even with the light is off.  It also does not matter where the sensitivity pot is set either.

When it is detecting pulses under test conditions (1 pulse every 2 seconds) I am getting wild kwh readings on the logs

22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.59875 kWh with 3 decimals of accuracy
-- | -- | -- | --
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.60844 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.61625 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.62937 kWh with 3 decimals of accuracy
22:57:29 | [D] | [light:035] | 'Red' Setting:
22:57:29 | [D] | [light:046] | State: OFF
22:57:29 | [D] | [sensor:127] | 'House - Power Consumption': Sending state 80357.14062 W with 0 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Power Consumption': Sending state 80357.14062 W with 0 decimals of accuracy
22:57:29 | [D] | [light:035] | 'Red' Setting:
22:57:29 | [D] | [light:046] | State: ON
22:57:29 | [D] | [sensor:127] | 'House - Daily Energy': Sending state 108.51303 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.65594 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Power Consumption': Sending state 0.00000 W with 0 decimals of accuracy
22:57:29 | [D] | [light:035] | 'Red' Setting:
22:57:29 | [D] | [sensor:127] | 'House - Daily Energy': Sending state 108.51303 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Daily Energy': Sending state 108.51303 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.67937 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.67937 kWh with 3 decimals of accuracy
22:57:29 | [D] | [light:035] | 'Red' Setting:
22:57:29 | [D] | [light:046] | State: OFF
22:57:29 | [D] | [light:046] | State: OFF
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.69032 kWh with 3 decimals of accuracy
22:57:29 | [D] | [light:035] | 'Red' Setting:
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.70062 kWh with 3 decimals of accuracy
22:57:29 | [D] | [light:035] | 'Red' Setting:
22:57:29 | [D] | [sensor:127] | 'House - Power Consumption': Sending state 375000.00000 W with 0 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Power Consumption': Sending state 375000.00000 W with 0 decimals of accuracy
22:57:29 | [D] | [light:035] | 'Red' Setting:
22:57:29 | [D] | [light:046] | State: ON
22:57:29 | [D] | [sensor:127] | 'House - Daily Energy': Sending state 108.52313 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.71438 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.71438 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.71938 kWh with 3 decimals of accuracy
22:57:29 | [D] | [light:035] | 'Red' Setting:
22:57:29 | [D] | [light:046] | State: OFF
22:57:29 | [D] | [sensor:127] | 'House - Power Consumption': Sending state 140625.00000 W with 0 decimals of accuracy
22:57:29 | [D] | [light:035] | 'Red' Setting:
22:57:29 | [D] | [light:046] | State: ON
22:57:29 | [D] | [sensor:127] | 'House - Daily Energy': Sending state 108.52602 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.72562 kWh with 3 decimals of accuracy
22:57:29 | [D] | [sensor:127] | 'House - Total Energy': Sending state 118.72562 kWh with 3 decimals of accuracy

My pulse rate on the meter is 3200 imp/kwh and the YAML file is set to this.

Current;y the ESP32 board is powered directly from my USB 3.0 port on my PC so should be getting enough power.

Any help will be appreciated as I am new to ESP32 boards and know nothing about them.

**Reply by klaasnicolaas** [reply]
Do you already use something like an [internal_filter](https://github.com/klaasnicolaas/home-assistant-glow/blob/main/home_assistant_glow.yaml#L166)?

**Reply by kallik** [reply]
Would it make sense to have the internal_filter as a toggle/set value in the integration similar to the pulse rate, also if you could specify as a parameter the max wattage of your power board / meter to get rid of erratic readings?

**Comment by markhss** [comment]
[home.txt](https://github.com/klaasnicolaas/home-assistant-glow/files/10118151/home.txt)
Thank you for your help
I have added an internal filter of 100ms

It has stopped erratic readings, but I am somehow getting reading of about 2000W of usage even when there is no signal for the light sensing LED to detect and if I turn the sensitivty pot down it does not detect any flashing at all and if I turn it all the way up it flashes like it is getting a signal without a signal being present.

I have attached the yaml file I am using and I have double checked the GPIO pins on my ESP32 and light sensing LED board

**Comment by markhss** [comment]
Thanks for everyones help

I had not read the instructions properly and I had not checked that my photodiode was the correct way round.

I have fixed that and everything is working correctly
